### PR TITLE
Creating shared base configuration

### PR DESCRIPTION
**Pull request by barndawgie**
I'd like to move to a model where configuration that's common across devices is unified in a single YAML file.

I will have a second PR which will update the individual per-device YAML files to inherit from this, but the base file needs to exist in the repo first since the package call will directly reference the file on github (thus it needs to exist).

### CODE REVIEW

File: econet_base.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  rx_pin: ${rx_pin}
+
+econet:
+  uart_id: uart_0
+  update_interval: ${econet_update_interval}
+
+binary_sensor:
+  - platform: econet
+    name: "Active Alerts"
+    sensor_datapoint: ALRMALRT
```
[the remark this answers is not in the document]
Comment (barndawgie) [reply econet_base.yaml:59]: If so I'll remove it here and move it to the per-device config.

[the remark this answers is not in the document]
Comment (barndawgie) [reply econet_base.yaml:59]: Actually, let me just back that out. Easy enough to add in a future PR.

File: econet_base.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  update_interval: ${econet_update_interval}
+
+sensor:
+  - platform: wifi_signal
+    name: "WiFi Signal Strength"
+    entity_category: "diagnostic"
+
+text_sensor:
+  - platform: version
+    name: "ESPHome Version"
```
Comment (tronikos) [inline econet_base.yaml:63]: Is this needed? It's already available as the firmware version under device info. If you have the ESPHome addon there is also an update entity. For me it's update.water_heater_firmware with:
```
auto_update: false
installed_version: 2023.8.3
in_progress: false
latest_version: 2023.9.0
release_summary: null
release_url: https://esphome.io/changelog/
skipped_version: null
title: ESPHome
device_class: firmware
entity_picture: https://brands.home-assistant.io/_/esphome/icon.png
friendly_name: Water Heater Firmware
supported_features: 1
```

**Review by tronikos — CHANGES_REQUESTED** [review]

File: econet_base.yaml (hunk 2 above)
Comment (barndawgie) [reply econet_base.yaml:63]: Not sure how I missed that field. Will remove it.

**Comment by tronikos** [comment]
Looking at your branch changes, it seems you will use a package https://github.com/esphome-econet/esphome-econet/compare/main...barndawgie:esphome-econet:packaging#diff-420cddf91d04924f10bc7dada68789b19b0b565fb741a070c4cea8583b39037cR8
How are you going to make it work for local?

**Comment by tronikos** [comment]
You can test it by modifying econet_base.yaml in a way that breaks it and then compile the example-local.yaml. I think it will succeed with your changes.

**Comment by barndawgie** [comment]
@tronikos  - Not sure what you mean by "for local."

For local code you can still use the `external_components_source` substitution. Or do you mean to run with only local configuration files?

**Comment by tronikos** [comment]
See my previous comment. How can we ensure when we run esphome compile example-local.yaml that all yaml and code used is from the local directory?
In addition, how can we ensure that the github workflows you recently added catch errors in econet_base.yaml?

**Comment by barndawgie** [comment]
I think this is solveable but let me go rework and resubmit.

**Pull request closed without merging**